{{cookiecutter.project_slug}}/.gitlab-ci.yml

```diff
@@ -0,0 +1,131 @@
+# templates
+# ------------
+
+.deploy-template:
+  image: alpine:latest
+  script:
+    - echo "Deploying to ${CI_ENVIRONMENT_NAME}"
+  variables:
+    GIT_STRATEGY: none  # prevents fetching repo (we only need the artifact here)
+
+# -------------
+# Pipeline Definition
+
+variables:
+  PIP_CACHE_DIR: $CI_PROJECT_DIR/.cache/pip
+  {%- if cookiecutter.package_manager == 'conda' %}
+  CONDA_PKGS_DIRS: $CI_PROJECT_DIR/.cache/conda
+  {%- elif cookiecutter.package_manager == 'poetry' %}
+  POETRY_VERSION: 1.0.5
+  POETRY_CACHE_DIR: $CI_PROJECT_DIR/.cache/poetry
+  {%- endif %}
+
+cache: &python_cache
+  paths:
+    - $PIP_CACHE_DIR
+    {%- if cookiecutter.package_manager == 'conda' %}
+    - $CONDA_PKGS_DIRS/*.tar.bz2
+    - $CONDA_PKGS_DIRS/urls.txt
+    {%- elif cookiecutter.package_manager == 'poetry' %}
+    - $POETRY_CACHE_DIR
+    {%- endif %}
+
+stages:
+  - build-wheel
+  - test-unit
+  - deploy-nonprod
+  - deploy-prod
+
+# Build and Test
+{% if cookiecutter.package_manager == 'poetry' -%}
+build-wheel:
+  image: python:3.7-stretch
+  stage: build-wheel
+  artifacts:
+    name: 'app-wheel'
+    paths:
+      - dist/{{ cookiecutter.module_name }}-*.whl
+    expire_in: 1 week
+  before_script:
+    - pip install poetry==$POETRY_VERSION
+  script:
+    - poetry build -f wheel
+{% else -%}
+build-wheel:
+  image: python:3.7-alpine
+  stage: build-wheel
+  artifacts:
+    name: 'app-wheel'
+    paths:
+      - dist/{{ cookiecutter.module_name }}-*.whl
+    expire_in: 1 week
+  script:
+    - python setup.py dist
```

Comment:
I think we need to install the dependencies first before we just build the package

Reply:
Are you sure? I Thought it's only relevant to have dependencies when installing the wheel. At least it worked fine like this when testing the pipeline.

Reply:
guess it depends on the type of build you're doing... some easy to spot issues like import errors can only be resolved if all dependencies are available. I'll look into this...